### PR TITLE
BUG: Fix+test division by negative zero

### PR DESCRIPTION
**Pull request by jbrockmendel**
Discovered while trying to reconcile Series vs Index arithmetic behavior.

A couple of other small misc changes ported from other branches, will be noted inline.

### CODE REVIEW

File: pandas/core/arrays/integer.py [hunk 1]
```diff
@@ -1,5 +1,4 @@
 import numbers
-import sys
```
Comment (jbrockmendel) [inline pandas/core/arrays/integer.py]: all edits in this file are in the "misc" category

File: pandas/tests/extension/test_categorical.py [hunk 2]
```diff
@@ -41,14 +41,19 @@ def dtype():
     return CategoricalDtype()
 
 
-@pytest.fixture
 def data():
-    """Length-100 array for this type.
```
Comment (jbrockmendel) [inline pandas/tests/extension/test_categorical.py]: All edits in this file are in the "misc" category

File: pandas/tests/io/pytables/test_pytables.py [hunk 3]
```diff
@@ -4339,6 +4339,7 @@ def test_store_datetime_mixed(self):
         df["d"] = ts.index[:3]
         self._check_roundtrip(df, tm.assert_frame_equal)
 
+    # FIXME: don't leave commented-out code
     # def test_cant_write_multiindex_table(self):
```
Comment (jbrockmendel) [inline pandas/tests/io/pytables/test_pytables.py]: misc category

File: setup.cfg [hunk 4]
```diff
@@ -84,6 +84,8 @@ plugins = Cython.Coverage
 [coverage:report]
 ignore_errors = False
 show_missing = True
+omit =
+    pandas/_version.py
 # Regexes for lines to exclude from consideration
```
Comment (jbrockmendel) [inline setup.cfg]: misc category, xref #26877

File: pandas/tests/extension/test_categorical.py [hunk 5]
```diff
@@ -41,14 +41,19 @@ def dtype():
     return CategoricalDtype()
 
 
-@pytest.fixture
```
Comment (jreback) [inline pandas/tests/extension/test_categorical.py]: pls don't change the fixture setup; these are very explict and match every other extension type.

**Review by jreback — CHANGES_REQUESTED** [review]
aside from a negative change in test_categorical ok

**Comment by jreback** [comment]
is this change user facing at all? or just test comparisons?

File: pandas/tests/extension/test_categorical.py (hunk 5 above)
Comment (jbrockmendel) [reply pandas/tests/extension/test_categorical.py]: ok, will change them all in a dedicated pass (as discussed at the sprint)

Comment (jreback) [reply pandas/tests/extension/test_categorical.py]: right if you change them *all* prob ok, but separate pass

**Comment by jbrockmendel** [comment]
> is this change user facing at all? or just test comparisons?

Yes, just barely:

```
np.random.seed(1)
idx = pd.Index(np.random.randn(5))

>>> idx // (idx * 0)

# master
Float64Index([inf, -inf, -inf, -inf, inf], dtype='float64')

# PR
Float64Index([inf, inf, inf, inf, inf], dtype='float64')
```

Similar for divmod.  truediv is unchanged.  Series is unchanged.

**Comment by jreback** [comment]
> > is this change user facing at all? or just test comparisons?
> 
> Yes, just barely:
> 
> ```
> np.random.seed(1)
> idx = pd.Index(np.random.randn(5))
> 
> >>> idx // (idx * 0)
> 
> # master
> Float64Index([inf, -inf, -inf, -inf, inf], dtype='float64')
> 
> # PR
> Float64Index([inf, inf, inf, inf, inf], dtype='float64')
> ```
> 
> Similar for divmod. truediv is unchanged. Series is unchanged.

ok that's fine then, wouldn't bother with a release note.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**